config/cluster/kind-cluster.yaml

```diff
@@ -0,0 +1,7 @@
+kind: Cluster
+apiVersion: kind.x-k8s.io/v1alpha4
+nodes:
+- role: control-plane
+networking:
+  ipFamily: dual
```

Comment:
Can you actually get dual IP in the pipeline? AFAIK GitHub doesn't support ipv6

Reply:
Yeah, the seems like that it is failing due to cluster not supporting Dual IP. Thinking on how to approach this.

Reply:
https://github.com/actions/runner-images/issues/668#issuecomment-1872487355

Maybe a way to get it working? Should look into it closely...

Reply:
I am updating the `kind create` step with the current config for `dual` IP, checking if it works.

Reply:
2 out of the 3 tests pass for IPFamily in kind cluster. Unable to figure out why the 3rd is [failing](https://github.com/nginxinc/nginx-gateway-fabric/actions/runs/10085057316/job/27885218042?pr=2238#step:16:79). 

I am unable to reproduce the issue locally but have asked @lucacome for some help

Passing tests
 - Service with IPv4 and IPv6 configured, when NGF is enabled on dual stack
 - Service with IPv4 configured, Service with IPv6 on the HTTPRoute , when NGF is enabled on IPv4 stack
Failing Tests
 - Service with IPv6 configured, Service with IPv4 error on the HTTPRoute,  when NGF is enabled on IPv6 stack